changelog/next/features/4527--precise-parsers.md

```diff
@@ -0,0 +1,8 @@
+The CEF, CSV, GELF, JSON, LEEF, Suricata, XSV, YAML, Zeek JSON parsers
+now properly adhere to the schema of the read data. Previously, parsers
+would merge heterogeneous input into a single, growing schema,
+inserting nulls for fields that did not exist in some events.
+
+These parsers now also support the `--schema`, `--selector` flags to
+parse their data according to some given schema, as well as various other
+flags to more precisely control their output schema.
```

Comment:
Every paragraph is a separate changelog entry, so this should also list all of them.